### PR TITLE
SUP-1693 Bind "s" to stop agent

### PR DESCRIPTION
**Pull request by jradtilbrook**

### CODE REVIEW

**Comment by jradtilbrook** [comment]
This is more or less working. There is a UI bug though that seems to not properly update the list (ie I can stop one agent and it gets stopped but the UI removes the wrong agent or it thinks there are still agents in the list)

**Comment by jradtilbrook** [comment]
For some reason the agent that gets stopped and should be removed from the list of agents is not getting removed. The index in the list that it occupies is set to nil. So the code still thinks the list is X long even though one of the items in that list is nil. I'm still trying to track down where that is being set to nil instead of outright removed

**Comment by jradtilbrook** [comment]
I've now got this working in tests but I don't think it actually works running it manually; agents never seem to load into the view

**Comment by jradtilbrook** [comment]
Scratch that, its running forever in CI for some reason, but it passes locally!!

**Comment by jradtilbrook** [comment]
I've marked this ready for review in case you can spot what the problem is with the tests. Or even if they pass/not pass locally for you could be helpful. Otherwise I think we should mark these tests as skipped for now so we can get this stop functionality merged, even in this state

**Comment by james2791** [comment]
Both also pass locally for me 🤔 

Something `HangUpException` in CI?

**Review by james2791 — CHANGES_REQUESTED** [review]

File: internal/agent/list.go [hunk 1]
```diff
@@ -109,16 +142,20 @@ func (m AgentListModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
...
 
+	if m, ok := msg.(Cmder); ok {
+		cmds = append(cmds, m.Cmd())
+	}
+
 	return m, tea.Batch(cmds...)
 }
 
 func (m AgentListModel) View() string {
-	return agentListStyle.Render(m.agentList.View())
```
[the remark this answers is not in the document]
Comment (jradtilbrook) [reply internal/agent/list.go:123]: Yeah I think its unnecessary

[the remark this answers is not in the document]
Comment (james2791) [reply internal/agent/list.go:123]: 👍

**Review by james2791 — APPROVED** [review]
:shipit: (3.x rebase/merge proceeding)

**Review by james2791 — APPROVED** [review]

**Pull request merged**